commit 6922f1be08111d889b0585b763b08f92d7a55e05
Author: Tianon Gravi <admwiggin@gmail.com>
Date:   Sat Feb 1 21:40:51 2014 -0700

    Remove reference to <linux/btrfs.h>, and instead use <btrfs/ioctl.h> like we're supposed to (from btrfs-progs)
    
    This fixes compilation issues when btrfs.h isn't available (because we just need the relevant structs, which for userspace programs are supposed to come from btrfs-progs instead of the kernel headers).
    
    Docker-DCO-1.1-Signed-off-by: Andrew Page <admwiggin@gmail.com> (github: tianon)

diff --git a/graphdriver/btrfs/btrfs.go b/graphdriver/btrfs/btrfs.go
index a50f11f..3d27909 100644
--- a/graphdriver/btrfs/btrfs.go
+++ b/graphdriver/btrfs/btrfs.go
@@ -4,15 +4,11 @@ package btrfs
 
 /*
 #include <stdlib.h>
-#include <sys/ioctl.h>
-#include <linux/fs.h>
-#include <errno.h>
-#include <sys/types.h>
 #include <dirent.h>
-#include <linux/btrfs.h>
-
+#include <btrfs/ioctl.h>
 */
 import "C"
+
 import (
 	"fmt"
 	"github.com/dotcloud/docker/graphdriver"
